[python] add axis delete to Experiment

bkmartinjr opened the pull request:
Add `Experiment.var_axis_delete` and `Experiment.obs_axis_delete` and their unit tests.

bkmartinjr commented on apis/python/src/tiledbsoma/_experiment.py:
```diff
@@ -98,3 +108,243 @@ def axis_query(
+    """
+    if isinstance(obj, DenseNDArray):
+        raise SOMAError(f"Delete operation not permitted on dense array: {name}")
+    if isinstance(obj, (DataFrame, SparseNDArray, GeometryDataFrame, PointCloudDataFrame)) and all(
+        obj.schema.get_field_index(col) >= 0 for col in join_on
+    ):
```
True, but I didn't want to duplicate the error checking already implemented in delete_cells.

That said, we can remove this easily and ignore the spatial arrays, at the cost of potential user confusion.  My sense is we need to raise a proactive error if the Experiment contains _any_ unsupported sub-array.

What do you recommend?

jp-dark replied:
I think we raise a `NotImplementedError` here if there is a `GeometryDataFrame`. I think it's unlikely anyone is using the `GeometryDataFrame` in the wild right now, but I don't want to corrupt an `Experiment` if they are.

aaronwolen reviewed (changes requested):
Looks great and the docstrings are all very clear/informative. Most of my suggestions are formatting-related.

bkmartinjr commented:
I believe I have addressed all of the issues:
1. formatting & copy editing in docstrings
2. added missing unit test
3. added silent filtering of any joinids outside the domain/shape of the target array

aaronwolen reviewed (approved):
Awesome, thank you!
